.github/workflows/main.yml

```diff
@@ -6,6 +6,38 @@ on:
     branches: [ master ]
 
 jobs:
+  sphinx:
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version:
+          - "3.7"
+          - "3.8"
+          - "3.9"
+          - "3.10"
+    runs-on: ubuntu-latest
```

Comment:
Why not run on a lighter linux distro?

Reply:
This is what GitHub makes available: https://github.com/actions/runner-images 
There are not really other linux distros available unless you self-host it, which is not needed.